tests/lib/location/CMakeLists.txt

```diff
@@ -44,6 +44,8 @@ cmock_handle(${ZEPHYR_BASE}/include/zephyr/net/wifi_mgmt.h
 cmock_handle(${ZEPHYR_BASE}/include/zephyr/device.h
 	     FUNC_EXCLUDE ".*DEVICE_DT_NAME_GET"
 	     WORD_EXCLUDE ".*device_visitor_callback_t.*")
+cmock_handle(${ZEPHYR_NRF_MODULE_DIR}/lib/location/scan_wifi.h
+	     FUNC_EXCLUDE ".*scan_wifi.*")
```

Comment:
This is not ideal as we are leaving some part of the code out of the tests. I'll see later if I'm able to find a better solution.

Reply:
Agreed, this was just to get  things moving.

Reply:
I think I've found a way around the issues causing` undefined reference to __device_dts_ord_12`. This was seen elsewhere too. I'm adding this close to the beginning of `location_test.c` (works for lwm2m_client_utils tests too):
```
#define DT_DRV_COMPAT nordic_wlan0
DEVICE_DT_INST_DEFINE(0, NULL, NULL, NULL, NULL, POST_KERNEL, 0, NULL);
```
I'll create a branch from these changes which you can cherry-pick and squash into this PR

Reply:
It's all good now!